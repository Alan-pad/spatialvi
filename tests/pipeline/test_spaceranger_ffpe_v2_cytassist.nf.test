nextflow_pipeline {
    name "Test full workflow including spaceranger"
    script "main.nf"
    tag "pipeline"

    test("spaceranger ffpe v2 cytassist (default `-profile test`)") {
        when {
            params {
                input = "https://raw.githubusercontent.com/fasterius/test-datasets/spatialtranscriptomics/testdata/human-brain-cancer-11-mm-capture-area-ffpe-2-standard_v2_ffpe_cytassist/samplesheet_spaceranger.csv"
                spaceranger_probeset = "https://raw.githubusercontent.com/nf-core/test-datasets/spatialtranscriptomics/testdata/human-brain-cancer-11-mm-capture-area-ffpe-2-standard_v2_ffpe_cytassist/outs/probe_set.csv"
                spaceranger_reference = "https://raw.githubusercontent.com/nf-core/test-datasets/spatialtranscriptomics/testdata/homo_sapiens_chr22_reference.tar.gz"
                st_preprocess_min_counts = 5
                st_preprocess_min_genes = 3
                outdir = "$outputDir"
            }
        }

        then {
            assertAll(

                // Workflow
                { assert workflow.success },
                { assert snapshot(UTILS.removeNextflowVersion("$outputDir")).match("software_versions") },

                // Data
                { assert file("$outputDir/CytAssist_11mm_FFPE_Human_Glioblastoma_2/data/st_adata_processed.h5ad").exists() },
                { assert file("$outputDir/CytAssist_11mm_FFPE_Human_Glioblastoma_2/data/st_adata_norm.h5ad").exists() },
                { assert file("$outputDir/CytAssist_11mm_FFPE_Human_Glioblastoma_2/data/st_adata_plain.h5ad").exists() },
                { assert file("$outputDir/CytAssist_11mm_FFPE_Human_Glioblastoma_2/data/st_adata_raw.h5ad").exists() },

                // Reports
                { assert
                { assert file("$outputDir/CytAssist_11mm_FFPE_Human_Glioblastoma_2/reports/st_qc_and_normalisation.html").text.contains("Distributions after filtering") },
                file("$outputDir/CytAssist_11mm_FFPE_Human_Glioblastoma_2/reports/st_clustering.html").text.contains("AnnData object is saved in the") },
                { assert file("$outputDir/CytAssist_11mm_FFPE_Human_Glioblastoma_2/reports/st_spatial_de.html").text.contains("Spatially variable genes") },

                // DEGs
                { assert file("$outputDir/CytAssist_11mm_FFPE_Human_Glioblastoma_2/degs/st_spatial_de.csv").exists() },

                // Space Ranger
                { assert file("$outputDir/CytAssist_11mm_FFPE_Human_Glioblastoma_2/outs/web_summary.html").exists() },
                { assert snapshot(
                    path("$outputDir/CytAssist_11mm_FFPE_Human_Glioblastoma_2/outs/raw_feature_bc_matrix.h5"),
                    path("$outputDir/CytAssist_11mm_FFPE_Human_Glioblastoma_2/outs/filtered_feature_bc_matrix.h5"),
                )},
                { assert file("$outputDir/CytAssist_11mm_FFPE_Human_Glioblastoma_2/outs/raw_feature_bc_matrix.h5").exists() },
                { assert file("$outputDir/CytAssist_11mm_FFPE_Human_Glioblastoma_2/outs/spatial/tissue_positions.csv").exists() },

                // MultiQC
                { assert file("$outputDir/multiqc/multiqc_report.html").text.contains("CytAssist_11mm_FFPE_Human_Glioblastoma_2_4")}
            )
        }
    }



}
