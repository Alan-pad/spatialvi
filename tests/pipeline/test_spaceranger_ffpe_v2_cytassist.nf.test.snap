{
    "nf_core_pipeline_software_mqc_versions.yml": {
        "content": [
            "{FASTQC={fastqc=0.12.1}, SPACERANGER_COUNT={spaceranger=2.1.0}, SPACERANGER_UNTAR_REFERENCE={untar=1.3}, ST_CLUSTERING={quarto=1.4.549, papermill=null}, ST_QUALITY_CONTROLS={quarto=1.4.549, papermill=null}, ST_READ_DATA={scanpy=1.9.8}, ST_SPATIALLY_VARIABLE_GENES={quarto=1.4.549, papermill=null}, UNTAR_SPACERANGER_INPUT={untar=1.3}, Workflow={nf-core/spatialtranscriptomics=v1.0dev}}"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-07T14:28:20.163729"
    }
}
